File: foobar.rb
puts "How many items of the Foobar pattern do you want to see?"

user_input = gets.chomp.to_i

number = 1

while number <= user_input
  if number % 3 == 0 && number % 5 == 0
    puts "Foobar"
  elsif number % 3 == 0
    puts "Foo"
  elsif number % 5 == 0
    puts "Bar"
  else
    puts number
  end
  number += 1
end

File: firehose/foobar.js
var readlineSync = require('readline-sync');

var input = readlineSync.question('How many elements of the Foobar pattern do you want to see? ');

for (var iteration = 1; iteration <= Number(input); iteration++) {
  if (iteration % 3 === 0 && iteration % 5 === 0) {
    console.log('Foobar');
  }
  else if (iteration % 3 === 0) {
    console.log('Foo');
  }
  else if (iteration % 5 === 0) {
    console.log('Bar');
  }
  else {
    console.log(iteration);
  }
}
